### DISCUSSION TITLE
Passing parameters from XYZGrid to XYZDialog

### DISCUSSION THREAD

**Discussion by JohnD832** [serenity-is/Serenity#7013] [Help]
Hi all

Just a heads up, I did trying Googling and searching the existing discussions and saw no results.

So I have the following `XYZGrid.ts`

![image](https://github.com/serenity-is/Serenity/assets/138233693/0e18ec25-ee73-4392-b78a-acbf085012a9)

Clicking that **Normal Shift** link,  opens up the `XYZDialog.ts`. The problem is, I want to pass some parameters to my dialog, the constructor for my dialog is like this:

```
constructor(public rotaTemplateId: number, public rotaTemplateOuId: number) {
    super();

    // do some stuff with those params
}
```

Right now, the first param comes in as an empty object `{}` and the second one comes in as `undefined`.

I want to pass values from the `XYZGrid.ts` to the `XYZDialog.ts`, how do I go about doing that?

In the `XYZColumns.cs`, I have the following:

```
namespace MyApp.Organisation.Columns
{
    using Serenity.ComponentModel;
    using System.ComponentModel;

    [ColumnsScript("Organisation.RotaTemplateShift")]
    [BasedOnRow(typeof(Entities.RotaTemplateShiftRow), CheckNames = true)]
    public class RotaTemplateShiftColumns
    {
        [Width(120)]
        [DisplayName("Icon Text")]
        public string RotaShiftTypeBoxIcon { get; set; }

        [Width(150)]
        [EditLink, DisplayName("Rota Shift Type")]
        [InlineRotaShiftTypeNameFormatter]
        public string RotaShiftTypeName { get; set; }

        [Width(150)]
        [DisplayName("Start Day")]
        [InlineRotaTemplateShiftStartDayFormatter]
        public int StartDay { get; set; }

        [Width(150)]
        [DisplayName("Is overtime override")]
        public bool IsOvertimeOverride { get; set; }

        [Width(200)]
        [DisplayName("Location")]
        public string RotaOuLocationStubName { get; set; }
    }
}
```

Notice the `[InlineRotaShiftTypeNameFormatter]` which in its `.ts` file has:

```
namespace MyApp.Organisation {

    @Serenity.Decorators.registerFormatter()
    export class InlinePeopleMyTeamSupportSupportedPersonFormatter
        implements Slick.Formatter, Serenity.IInitializeColumn {

        format(ctx: Slick.FormatterContext): string {

            let output = `<a href="#" style="text-decoration: underline;" class="supported-person">${ctx.value}</a>`;
            return output;
        }

        initializeColumn(column: Slick.Column): void {
        }
    }
}
```

Any help would be greatly appreciated

**Comment by stixoffire** [comment]
Search for a discussion and comment by John Ranger .. there is a method to do this .. I am in my office right now and can't look it up - but I also use this method to transfer between forms - setup a property on your form and then set the value.

Don't overthink it , it is something simple like myform.property = xyzvalue if I have a moment I will try and find it. I think John ranger had it in the WIKI as well.
